django/core/management/commands/loaddata.py

```diff
@@ -128,7 +128,16 @@ def load_label(self, fixture_label):
             fixture = open_method(fixture_file, mode)
             try:
                 self.fixture_count += 1
-                objects_in_fixture = 0
+                if self.verbosity == 3:
+                    class ObjectsInFixtureNotifier(int):
+                        def __iadd__(self, other):
+                            self = ObjectsInFixtureNotifier(self + other)
+                            self.stdout.write('Processed %i object(s)\r' % self)
+                            return self
+                    ObjectsInFixtureNotifier.stdout = self.stdout
+                    objects_in_fixture = ObjectsInFixtureNotifier(0)
+                else:
+                    objects_in_fixture = 0
```

Comment:
IMHO this is a bit over-engineered.

I get that you want to avoid checking `if self.verbosity == 3` on each iteration of the `for obj in objects` block below but I think the performance hit would be negligible compared to all operations taking place during unserialization of an object.

Do you think it would make sense adding those checks below instead?